discovery/manager_test.go

```diff
@@ -801,43 +807,32 @@ scrape_configs:
 // removing all targets from the static_configs sends an update with empty targetGroups.
 // This is required to signal the receiver that this target set has no current targets.
 func TestTargetSetRecreatesEmptyStaticConfigs(t *testing.T) {
-	cfg := &config.Config{}
-
-	sOne := `
-scrape_configs:
- - job_name: 'prometheus'
-   static_configs:
-   - targets: ["foo:9090"]
-`
-	if err := yaml.UnmarshalStrict([]byte(sOne), cfg); err != nil {
-		t.Fatalf("Unable to load YAML config sOne: %s", err)
-	}
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	discoveryManager := NewManager(ctx, log.NewNopLogger())
 	discoveryManager.updatert = 100 * time.Millisecond
 	go discoveryManager.Run()
 
 	c := make(map[string]sd_config.ServiceDiscoveryConfig)
-	for _, v := range cfg.ScrapeConfigs {
-		c[v.JobName] = v.ServiceDiscoveryConfig
+	c["prometheus"] = sd_config.ServiceDiscoveryConfig{
+		StaticConfigs: []*targetgroup.Group{
+			&targetgroup.Group{
```

Comment:
same here, this matches the target group on lines 763 and 789, it could be defined once outside the tests

Reply:
you mean a common variable for tow tests ?
```go
var TG = targetgroup.Group {
    // targets: "foo:9090"
}

func TestA() {
    // use TG here
}
func TestB() {
    // use TG here
}
```
May not be a good idea.

Reply:
Something like this 
```
var fooTarget = &targetgroup.Group{
    Source: "0",
    Targets: []model.LabelSet{
        model.LabelSet{
	    "__address__": "foo:9090",
        },
    },
}

...

func TestTargetSetRecreatesEmptyStaticConfigs(t *testing.T) {
    // some setup
    c["prometheus"] = sd_config.ServiceDiscoveryConfig{
        StaticConfigs: []*targetgroup.Group{
            &fooTg,
        },
    }

    // check that the TG is present
    // change c["prometheus"] to an empty target group
    // check that length of targetgroups in discoveryManager is now 0
}

Reply:
And also use `fooTarget` here ?
```go
func TestTargetSetRecreatesTargetGroupsEveryRun(t *testing.T) {
// ...
	c["prometheus"] = sd_config.ServiceDiscoveryConfig{
		StaticConfigs: []*targetgroup.Group{

			&fooTarget,  // <----

			&targetgroup.Group{
				Source: "1",
				Targets: []model.LabelSet{
					model.LabelSet{
						"__address__": "bar:9090",
					},
				},
			},
		},
	}
// ...
}
```
If yes, I don't think it's a good idea. The point is, these two tests will depend on the same target. What if I change it for `TestTargetSetRecreatesEmptyStaticConfigs`? I may break another test.

Plus, we have a lot duplication of this kind, I think they are reasonable.
e.g.
https://github.com/prometheus/prometheus/blob/08c55c119f39093e18b2bb9cba5c5619dc4ea0e1/discovery/manager_test.go#L111-L115
and
https://github.com/prometheus/prometheus/blob/08c55c119f39093e18b2bb9cba5c5619dc4ea0e1/discovery/manager_test.go#L141-L145

Reply:
Apparently, we have some different ideas here, @beorn7 can you take a review please ?

Reply:
Personally the duplication in tests doesn't bother me so much.

Reply:
I don't feel strongly about changing it. This is good to go imo cc @simonpasquier